### PR TITLE
docs(MADR): use MeshHTTPRoute as spec.targetRef.kind

### PR DESCRIPTION
**Pull request by lobkovilya**
### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues -- https://github.com/kumahq/kuma/issues/5867
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — APPROVED** [review]
LGTM

**Review by bartsmykla — APPROVED** [review]

File: docs/madr/decisions/028-meshhttproute-as-targetref.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  http:
+    numRetries: 500
+    retryOn:
+      - "5xx"
+```
+
+### Inspect API
+
+In Kuma GUI in `To` column we have to display not only the outbound's name, but also `rule[].matches` list, because
+this list uniquely identifies routes in Envoy.
```
Comment (lahabana) [inline docs/madr/decisions/028-meshhttproute-as-targetref.md:287]: Can you show an example of this new payload?

**Review by lahabana — COMMENTED** [review]
So to resolve which policies apply to this DPP you'll first need to figure out which meshHTTPRoute apply and then be able to compute for all the other policies?

I think one of the other option was the use `MeshHTTPRoute` as an option in the `to` section. I don't see in this MADR the explanation why this was rejected.

Also more generally MADR are history documents so it'd be nice to make it explicit why alternatives are rejected (useful to go back and remind ourselves why we went in a direction and not another)

**Comment by lobkovilya** [comment]
@lahabana I agree, added section for the "top-level vs to[]" decision

**Review by lahabana — APPROVED** [review]

**Pull request merged**